### DISCUSSION TITLE
How to  override the seeding of default user name and password?

### DISCUSSION THREAD

**Discussion by mrs01dev** [abpframework/abp#6909] [Q&A]
The  Hidden class "IdentityDataSeedContributor" in "modules/identity"  seeds default user name and password .
It not a wisdom to  deploy application with a security breaches [ default user name and password].
How to  override the seeding of default user name and password?

**Comment by maliming** [comment]
hi

You can check: https://github.com/abpframework/abp/issues/3088 and https://github.com/abpframework/abp/issues/2458

**Comment by mrs01dev** [comment]
Thanks , my final code for seeding  User Name , Password   and Tenants 
1- modify  MyProjectNameDbMigrationService.cs 
 public class ***DbMigrationService : ITransientDependency
    {
        public ILogger<***DbMigrationService> Logger { get; set; }

        private readonly IDataSeeder _dataSeeder;
        private readonly IEnumerable<ITSPDbSchemaMigrator> _dbSchemaMigrators;
        private readonly ITenantRepository _tenantRepository;
        private readonly ITenantManager _tenantManager;
        private readonly ICurrentTenant _currentTenant;
        private readonly IConfiguration _configuration;
        public ***DbMigrationService(
            IDataSeeder dataSeeder,
            IEnumerable<ITSPDbSchemaMigrator> dbSchemaMigrators,
            ITenantRepository tenantRepository,
            ITenantManager tenantManager,
            ICurrentTenant currentTenant,
            IConfiguration configuration)
        {
            _dataSeeder = dataSeeder;
            _dbSchemaMigrators = dbSchemaMigrators;
            _tenantRepository = tenantRepository;
            _tenantManager = tenantManager;
            _currentTenant = currentTenant;
            _configuration = configuration;
            Logger = NullLogger<TSPDbMigrationService>.Instance;
        }

        public async Task MigrateAsync()
        {
            Logger.LogInformation("Started database migrations...");

            await MigrateDatabaseSchemaAsync();
            await SeedTenantAsync();
            await SeedDataAsync();

            Logger.LogInformation($"Successfully completed host database migrations.");

            var tenants = await _tenantRepository.GetListAsync(includeDetails: true);

            var migratedDatabaseSchemas = new HashSet<string>();
            foreach (var tenant in tenants)
            {
                using (_currentTenant.Change(tenant.Id))
                {
                    if (tenant.ConnectionStrings.Any())
                    {
                        var tenantConnectionStrings = tenant.ConnectionStrings
                            .Select(x => x.Value)
                            .ToList();

                        if (!migratedDatabaseSchemas.IsSupersetOf(tenantConnectionStrings))
                        {
                            await MigrateDatabaseSchemaAsync(tenant);

                            migratedDatabaseSchemas.AddIfNotContains(tenantConnectionStrings);
                        }
                    }

                    await SeedDataAsync(tenant);
                }

                Logger.LogInformation($"Successfully completed {tenant.Name} tenant database migrations.");
            }

            Logger.LogInformation("Successfully completed database migrations.");
        }

        private async Task MigrateDatabaseSchemaAsync(Tenant tenant = null)
        {
            Logger.LogInformation(
                $"Migrating schema for {(tenant == null ? "host" : tenant.Name + " tenant")} database...");

            foreach (var migrator in _dbSchemaMigrators)
            {
                await migrator.MigrateAsync();
            }
        }

        private async Task SeedDataAsync(Tenant tenant = null)
        {
            Logger.LogInformation($"Executing {(tenant == null ? "host" : tenant.Name + " tenant")} database seed...");
            string userEmail = string.Empty;
            string userPassword = string.Empty;
            IConfigurationSection userConfigurationSection = _configuration.GetSection("AdminUserIdentity");
            if (userConfigurationSection!= null)
            {
                userEmail = userConfigurationSection["Email"] ?? "admin@abp.io";
                userPassword = userConfigurationSection["Password"] ?? "1q2w3E*";
            }
            else
            {
                userEmail= "admin@abp.io";
                userPassword = "1q2w3E*";
            }
            // change user name by prefixing with tenantName
            if (tenant != null)
            {
                userEmail = tenant.Name + "_"+ userEmail;
                await _dataSeeder.SeedAsync(
                new DataSeedContext(tenant.Id)
            .WithProperty("AdminEmail", userEmail)
            .WithProperty("AdminPassword", userPassword)
            );

            }
            else
            {
                
                await _dataSeeder.SeedAsync(
               new DataSeedContext(null)
           .WithProperty("AdminEmail", userEmail)
           .WithProperty("AdminPassword", userPassword)
           );

            }


        }

        private async Task SeedTenantAsync()
        {
            List<string> tenantNameslist =new List<string>();

            var tenantConfigurationSection = _configuration.GetSection("TenantData:TenantNameslist");  
            if (tenantConfigurationSection != null)
            {
                 tenantNameslist = tenantConfigurationSection.Get<List<string>>();
               

            }
          
            tenantNameslist.Sort();


            foreach (string tenantName in tenantNameslist)
            {
                Tenant tenant = await _tenantManager.CreateAsync(tenantName);
             
               await  _tenantRepository.InsertAsync(tenant,true);


            }

        }
    }

2- add to appsettings in ***DbMigrator project
"AdminUserIdentity": {
    "Email": "admin@abp.io",
    "Password": "1q2w3E*"
  },
  "TenantData": {

    "TenantNameslist": [

      "Tokyo",
      "London",
      "Cairo",
      "Astana",
      "Berlin",
      "Washington",
    ]
    
  }
